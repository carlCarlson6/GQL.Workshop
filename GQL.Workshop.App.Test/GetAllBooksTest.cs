using System.Threading.Tasks;
using Snapshooter.Xunit;
using FluentAssertions;
using Xunit;

namespace GQL.Workshop.App.Test;

public class GetAllBooksTest : BaseApiTest
{
    [Fact]
    public async Task GetAllBooks()
    {
        // Given
        var client = GivenTestHost().GetGqlClient();

        // When
        var result = await client.AllBooks.ExecuteAsync();

        // Then
        result.Should().MatchSnapshot();
    }
}
