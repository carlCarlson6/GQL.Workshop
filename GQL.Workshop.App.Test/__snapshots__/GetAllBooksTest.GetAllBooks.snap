{
  "Data": {
    "Books": [
      {
        "Title": "Is graphql better than rest?"
      },
      {
        "Title": "The succss of Colba"
      }
    ]
  },
  "DataInfo": {
    "Books": [
      {
        "Name": "Book",
        "Value": "0dacb47e-2e37-459b-bc25-b9ab58bf89ee"
      },
      {
        "Name": "Book",
        "Value": "951c4b62-d33e-4687-bff4-397816c0860d"
      }
    ],
    "EntityIds": [
      {
        "Name": "Book",
        "Value": "0dacb47e-2e37-459b-bc25-b9ab58bf89ee"
      },
      {
        "Name": "Book",
        "Value": "951c4b62-d33e-4687-bff4-397816c0860d"
      }
    ],
    "Version": 1
  },
  "DataFactory": {},
  "Errors": [],
  "Extensions": {},
  "ContextData": {}
}
